dataset.go

```diff
@@ -274,6 +279,46 @@ func (set dataset) Equal(other Data) bool {
 	return true
 }
 
+// see Data.Compare
+// Compare compares given data to this data, row by row.
+// Compare set and the otherSet, column by column, and calculate final result
+// for each row by merging columns comparison results.
+func (set dataset) Compare(other Data) ([]compare.Comparison, error) {
+	otherSet, ok := other.(Dataset)
+	if !ok {
+		return nil, ErrMismatchTypes
+	}
+	if set.Width() != otherSet.Width() {
+		return nil, errCompareDiffTypes
+	}
+
+	// The 1st comparison iteration, doesn't need to merge the comparison results,
+	// because it's the only existing batch.
```

Comment:
lets avoid the word batch in data/dataset implementations :)
what do you think about:
 ```suggestion
	// prepare first compare results, to be merged with future comparison results
```